compiler/packages/babel-plugin-react-compiler/src/SSA/EliminateRedundantPhi.ts

```diff
@@ -141,6 +143,23 @@ export function eliminateRedundantPhi(
      * have already propagated forwards since we visit in reverse postorder.
      */
   } while (rewrites.size > size && hasBackEdge);
+
+  if (DEBUG) {
+    for (const [, block] of ir.blocks) {
+      for (const phi of block.phis) {
+        CompilerError.invariant(!rewrites.has(phi.place.identifier), {
+          reason: '[EliminateRedundantPhis]: rewrite not complete',
+          loc: phi.place.loc,
+        });
+        for (const [, operand] of phi.operands) {
+          CompilerError.invariant(!rewrites.has(operand.identifier), {
+            reason: '[EliminateRedundantPhis]: rewrite not complete',
+            loc: phi.place.loc,
+          });
+        }
+      }
+    }
+  }
```

Comment:
Happy to remove or gate behind a compiler flag, I ended up adding this to debug an issue (since fixed) in EnterSSA.

This checks that once we're done rewriting, no phi or phi operands should have pending rewrites